go/staking/tests/tester.go

```diff
@@ -326,25 +397,26 @@ func testEscrowEx(
 		t.Fatalf("failed to receive escrow event")
 	}
 
-	_ = srcAcc.General.Balance.Sub(&escrow.Tokens)
+	currentTotalShares := dstAcc.Escrow.Active.TotalShares.Clone()
+	_ = dstAcc.Escrow.Active.Deposit(currentTotalShares, &srcAcc.General.Balance, &escrow.Tokens)
+
 	newSrcAcc, err := backend.AccountInfo(context.Background(), &api.OwnerQuery{Owner: srcID, Height: consensusAPI.HeightLatest})
 	require.NoError(err, "src: AccountInfo - after")
 	require.Equal(srcAcc.General.Balance, newSrcAcc.General.Balance, "src: general balance - after")
 	if !srcID.Equal(dstID) {
-		require.True(newSrcAcc.Escrow.Active.Balance.IsZero(), "src: active escrow balance == 0 - after")
+		require.Equal(state.srcAccountEscrowActiveBalance, newSrcAcc.Escrow.Active.Balance, "src: active escrow balance unchanged - after")
 		require.True(newSrcAcc.Escrow.Debonding.Balance.IsZero(), "src: debonding escrow balance == 0 - after")
 	}
 	require.Equal(tx.Nonce+1, newSrcAcc.General.Nonce, "src: nonce - after")
 
-	_ = dstAcc.Escrow.Active.Balance.Add(&escrow.Tokens)
 	newDstAcc, err := backend.AccountInfo(context.Background(), &api.OwnerQuery{Owner: dstID, Height: consensusAPI.HeightLatest})
 	require.NoError(err, "dst: AccountInfo - after")
 	if !srcID.Equal(dstID) {
 		require.Equal(dstAcc.General.Balance, newDstAcc.General.Balance, "dst: general balance - after")
 		require.Equal(dstAcc.General.Nonce, newDstAcc.General.Nonce, "dst: nonce - after")
 	}
 	require.Equal(dstAcc.Escrow.Active.Balance, newDstAcc.Escrow.Active.Balance, "dst: active escrow balance - after")
-	require.Equal(dstAcc.Escrow.Active.Balance, newDstAcc.Escrow.Active.TotalShares, "dst: active escrow total shares - after")
+	require.Equal(dstAcc.Escrow.Active.TotalShares, newDstAcc.Escrow.Active.TotalShares, "dst: active escrow total shares - after")
```

Comment:
Indeed, since tokens and shares don't map 1:1, I updated the code to check `TotalShares` instead of `Balance` when checking the amount of shares in the account.